webapp/src/main/webapp/src/app/groups/groups.component.ts

```diff
@@ -1,85 +1,109 @@
-import { Component, OnInit } from "@angular/core";
+import { Component, OnInit, OnDestroy } from "@angular/core";
 import { GroupFilterOptions } from "./model/group-filter-options.model";
 import { ActivatedRoute, Router } from "@angular/router";
 import { GroupService } from "./groups.service";
 import { GroupQuery } from "./model/group-query.model";
 import { Group } from "./model/group.model";
-import { UserService } from "../user/user.service";
 import { DeleteGroupModalComponent } from "./delete-group.modal";
 import { BsModalService, BsModalRef } from "ngx-bootstrap";
+import { Subscription } from "rxjs";
 
 @Component({
   selector: 'groups',
   templateUrl: './groups.component.html'
 })
-export class GroupsComponent implements OnInit {
+export class GroupsComponent implements OnInit, OnDestroy {
+
   filterOptions: GroupFilterOptions;
-  groups: Group[];
-  filteredGroups: Group[];
   query: GroupQuery;
   searchTerm: string = '';
-  bsModalRef: BsModalRef;
   schoolDropdownOptions: any[];
+  filteredGroups: Group[];
+  private _groups: Group[];
+  private _modalSubscriptions: Subscription[] = [];
 
   constructor(private route: ActivatedRoute,
               private router: Router,
               private service: GroupService,
               private modalService: BsModalService) {
   }
 
+  get groups(): Group[] {
+    return this._groups;
+  }
+
+  set groups(groups: Group[]) {
+    this._groups = groups;
+    this.updateFilteredGroups();
```

Comment:
every time the groups update the filtered selection needs to update so it seemed logical to put this here.